website/api/zkvm/input-output.md

```diff
@@ -0,0 +1,274 @@
+into consideration that it _is_ included in the [session], and therefore any
+party that has access to it will also have access to the private data.
+
+## File Descriptors and I/O
+
+In the zkVM, private I/O happens through file descriptors. A file descriptor is
+a unique identifier that represents an open file or input/output resource in a
+computer program. It serves as an abstract representation of a file or device,
+enabling the program to read from or write to it. To perform input and output
+operations in the zkVM, we need to specify the file descriptors to work as the
+input and output. This allows the program to read input data from a specific
+source and write output data to a specific destination. By specifying the file
+descriptors, the program can establish connections with the desired input/output
+resources and perform the necessary operations.
+
+By default, the zkVM provides four file descriptors for I/O operations:
+
+- [`stdin`][stdin]: The standard input file descriptor. It is used to read input
+  data from the host to the guest.
+- [`stdout`][stdout]: The standard output file descriptor. It is used to write output
+  data from the guest to the host.
+- [`stderr`][stderr]: The standard error file descriptor. It is used to write error from the guest to the host.
+- [`journal`][journal]: The journal file descriptor. It is used to write public
+  data.
+
+For the [`stdin`][stdin], [`stdout`][stdout], and [`journal`][journal] file
+descriptors, there are some helper methods to read and write data, that we'll
+cover in the next sections.
+
+:::info
+It's important to highlight that from those file descriptors, only the [`journal`][journal] has a public nature. All the other file descriptors are private and will not be part of the proof.
+:::
+
+## Handling Inputs
+
+The [`stdin`][stdin] file descriptor is used to send input data from the host to
+the guest. The zkVM provides some convenience methods to read data from this
+file descriptor.
+
+- [`env::read`][read_method]: Reads a value from the `stdin` file descriptor.
+- [`env::read_slice`][read_slice_method]: Reads a slice from the `stdin` file
+  descriptor.
+
+For example, in the host program, we could:
+
+`/src/main.rs`
+
+```rust
+use risc0_zkvm::ExecutorEnv;
+
+fn main() {
+  let env = ExecutorEnv::builder()
+              // Write a single value to [stdin]
+              .write(42u64)
+              .unwrap()
+              // Write a slice to [stdin]
+              .write_slice(&[1u8, 2u8, 3u8])
+              .build();
+
+  // Do something with the ExecutorEnv...
+}
+```
+
+And in the guest program, we could:
+
+`/methods/guest/src/main.rs`
+
+```rust
+#![no_main]
+use risc0_zkvm::guest::env;
+risc0_zkvm::guest::entry!(main);
+
+fn main() {
+  // Read a single value from the standard input
+  let input: u64 = env::read()?;
+
+  // Read a slice from the standard input
+  let mut buffer = [0u8; 32];
+  env::read_slice(&mut buffer)?;
```

Comment:
If this code is used with the host code above, it would fail because it tries to read more data than is written. Since we are presenting both together, I think the implication is that they should work together.